src/dtx/dtx_srv.c

```diff
@@ -219,12 +219,19 @@ dtx_handler(crt_rpc_t *rpc)
 
 		rc = vos_dtx_check(cont->sc_hdl, din->di_dtx_array.ca_arrays,
 				   NULL, NULL, NULL, NULL, false);
-		if (rc == DTX_ST_INITED)
+		if (rc == DTX_ST_INITED) {
 			/* For DTX_CHECK, non-ready one is equal to non-exist. Do not directly
 			 * return 'DTX_ST_INITED' to avoid interoperability trouble if related
 			 * request is from old server.
 			 */
 			rc = -DER_NONEXIST;
+		} else if (rc == -DER_INPROGRESS && !dtx_cont_opened(cont)) {
+			/* Trigger DTX re-index for subsequent (retry) DTX_CHECK. */
+			rc1 = start_dtx_reindex_ult(cont);
```

Comment:
Hmm, should it stop retrying even after dtx table is loaded?

Reply:
I do not think so, because object rebuild/DTX resync do not depends on the container start/stop. Here, if we do not retry, then DTX resync on remote engine will fail, that will cause the subsequent object rebuild failure.

Reply:
if the dtx entry can not be found even after loading the pmem entry, what is the point of keep retrying?  I think the caller needs to be notified in this case. IMHO.

Reply:
If the DTX reindexed, then vos_dtx_check() will not return -DER_INPROGRESS any longer, then will no retry any more.

Reply:
ah, yes, vc_cmt_dtx_indexed is set. I see. thanks.